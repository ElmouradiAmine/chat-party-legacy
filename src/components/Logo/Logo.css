.logo {
  display: flex;
  align-items: center;
}

.logo__image {
  height: 4.2rem;
  margin-right: 0.8rem;
}

.logo__title {
  font-family: 'Pacifico', cursive;
  color: var(--color-font-primary);
  font-size: 2.4rem;
}

.logo__title--colored {
  margin-left: 0.8rem;
  color: #ffbc00;
}

.logo__version {
  font-size: 1.3rem;
}
