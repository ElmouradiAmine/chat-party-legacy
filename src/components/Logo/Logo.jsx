import React from 'react';
import PropTypes from 'prop-types';

import logo from '../../assets/logo.png';
import './Logo.css';

const Logo = ({ className }) => (
  <figure className={`logo ${className}`}>
    <img
      src={logo}
      alt="logo chat-pary linux penguin"
      className="logo__image"
    />
    <h1 className="logo__title">
      Chat
      <span className="logo__title--colored">Party</span>
      <sub className="logo__version">v4.0.0</sub>
    </h1>
  </figure>
);

Logo.propTypes = {
  className: PropTypes.string,
};

Logo.defaultProps = {
  className: '',
};

export default Logo;
