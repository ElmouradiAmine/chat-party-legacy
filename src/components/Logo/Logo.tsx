import React from 'react';

import logo from '../../assets/img/logo.png';

import './Logo.css';

interface Props {
  className?: string;
}
const Logo: React.FC<Props> = ({ className = '' }) => (
  <figure className={`logo ${className}`}>
    <img src={logo} alt="logo chat-pary linux penguin" className="logo__image" />
    <h1 className="logo__title">
      Chat
      <span className="logo__title--colored">
        Party
      </span>
      <sub className="logo__version">v4.0.0</sub>
    </h1>
  </figure>
);

export default Logo;
