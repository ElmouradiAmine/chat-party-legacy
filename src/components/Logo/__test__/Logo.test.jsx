import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Logo from '../Logo';

afterEach(cleanup);
describe('Logo component', () => {
  const  className = 'className';
  
  test('renders correctly', () => {
    render(<Logo />);
  });

  test('renders correctly with props', () => {
    const { container } = render(<Logo className={className} />);
    expect(container).not.toBeNull();
    expect(container.firstElementChild?.classList).toContain(className);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Logo className={className} />);
    expect(tree).toMatchSnapshot();
  });
});