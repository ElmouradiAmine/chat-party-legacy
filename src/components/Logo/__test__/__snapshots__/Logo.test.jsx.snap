// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component matches snapshot 1`] = `
<figure
  className="logo className"
>
  <img
    alt="logo chat-pary linux penguin"
    className="logo__image"
    src="logo.png"
  />
  <h1
    className="logo__title"
  >
    Chat
    <span
      className="logo__title--colored"
    >
      Party
    </span>
    <sub
      className="logo__version"
    >
      v4.0.0
    </sub>
  </h1>
</figure>
`;
